controllers/chat/moveToAnon.js

```diff
@@ -34,6 +34,17 @@ const moveToAnon = async (req, res) => {
   if (!members.includes(req.userId)) members.push(req.userId);
 
   const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+
+  const filter = {id: oldChannelId};
+  const sort = {last_message_at: -1};
+  const oldChannelDetail = await client.queryChannels(filter, sort);
+  //req.userId always anon since the token are using anon token
+  const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
+  const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+    (user) => user.user_id !== mySignedId
+  )[0];
+  const detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is querying the old channel details and finding the previous target user. However, there's no error handling for the case where the old channel or the previous target user doesn't exist. This could lead to a potential issue if either of these are not found.

```diff
+  let oldChannelDetail;
+  let detailPrevTargetUser;
+  try {
+    oldChannelDetail = await client.queryChannels(filter, sort);
+    const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+      (user) => user.user_id !== mySignedId
+    )[0];
+    detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);
+  } catch (error) {
+    // Handle the error appropriately
+    console.error(error);
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->